### PR TITLE
fix: finish bucket file coverage

### PR DESCRIPTION
**Pull request by Skillz4Killz**

### CODE REVIEW

File: packages/utils/tests/bucket.spec.ts [hunk 1]
```diff
@@ -169,5 +169,28 @@ describe('bucket.ts', () => {
...
+    it("Don't process queue twice", () => {
+      const bucket = new LeakyBucket({
+        max: 1,
+        refillInterval: 500,
+        refillAmount: 1,
+      })
+      // fake processing
+      bucket.processing = true;
+      // request when already processing
+      bucket.processQueue();
```
Comment (H01001000) [inline packages/utils/tests/bucket.spec.ts:193]: missing a check here

File: packages/utils/tests/bucket.spec.ts [hunk 2]
```diff
@@ -169,5 +169,28 @@ describe('bucket.ts', () => {
...
+    it("remaining is 0 when used too many", () => {
+      const bucket = new LeakyBucket({
+        max: 1,
+        refillInterval: 500,
+        refillAmount: 1,
+      })
+      // max is < used
+      bucket.used = 2;
+      expect(bucket.remaining).equals(0);
+    })
```
Comment (H01001000) [inline packages/utils/tests/bucket.spec.ts:182]:
```suggestion
  describe('remaining', () => {
    it("should be 0 even used too many", () => {
      const bucket = new LeakyBucket({
        max: 1,
        refillInterval: 500,
        refillAmount: 1,
      })
      // max is < used
      bucket.used = 2;
      expect(bucket.remaining).equals(0);
    })
  })
```

**Review by H01001000 — CHANGES_REQUESTED** [review]

File: packages/utils/tests/bucket.spec.ts (hunk 1 above)
Comment (Skillz4Killz) [reply packages/utils/tests/bucket.spec.ts:193]: what check?

Comment (H01001000) [reply packages/utils/tests/bucket.spec.ts:195]: how do you check if it processes twice?

Comment (Skillz4Killz) [reply packages/utils/tests/bucket.spec.ts:195]: it doesn't process twice, the second request when already processing is true should just cancel out, no way to check that

**Pull request merged**